src/Device/Driver/OpenVario.cpp

```diff
@@ -46,65 +46,162 @@ class OpenVarioDevice : public AbstractDevice {
   bool PutBugs(double bugs, OperationEnvironment &env) override;
   void OnCalculatedUpdate(const MoreData &basic,
                   const DerivedInfo &calculated) override;
-  static bool POV(NMEAInputLine &line, NMEAInfo &info);
+private:
+  bool POV(NMEAInputLine &line, NMEAInfo &info);
+  bool ComposeWrite(const char p_type,PolarCoefficients &p,
+  		  OperationEnvironment &env);
+  bool PutIdealPolar(const DerivedInfo &calculated,
+  		  OperationEnvironment &env);
+  bool PutRealPolar(const DerivedInfo &calculated,
+  		  OperationEnvironment &env);
+  bool RepeatBugs(OperationEnvironment &env);
+  bool RepeatBallast(OperationEnvironment &env);
+  bool RepeatMacCready(OperationEnvironment &env);
+  bool IsEqual(const PolarCoefficients &p1, const PolarCoefficients &p2); // all 3 coeffs equal between the 2 polars
+  void Copy(PolarCoefficients &dest, const PolarCoefficients &src); // copy the 3 coffeicients from src to dest
+
+  // remember the settings to be able to repeat the most recent values upon request
+  double _bugs = 1;
+  bool   _bugs_valid = false;
+  double _mc = 1;
+  bool   _mc_valid = false;
+  double _overload = 1;
+  bool   _overload_valid = false;
+  PolarCoefficients _ideal_polar; 
+  bool _ideal_polar_valid = false;
+  PolarCoefficients _real_polar; 
+  bool _real_polar_valid = false;
 };
 
+bool
+OpenVarioDevice::IsEqual(const PolarCoefficients &p1, const PolarCoefficients &p2)
+{
+  if (p1.a != p2.a) return false;
+  if (p1.b != p2.b) return false;
+  if (p1.c != p2.c) return false;
+  return true;
+}
+
+void
+OpenVarioDevice::Copy(PolarCoefficients &dest, const PolarCoefficients &src)
+{
+  dest.a = src.a;
+  dest.b = src.b;
+  dest.c = src.c;
+}
+
+bool
+OpenVarioDevice::ComposeWrite(const char p_type,PolarCoefficients &p,
+  OperationEnvironment &env)
+{
+  if (!EnableNMEA(env))
+    return false;
+
+  char buffer[50];
+  // Compose Polar String
+  // TODO does the receiver understand %e format? Or do we have to use %f, even if
+  //      the number of significant digits is less then 3.
+  sprintf(buffer,"POV,C,%cPO,%e,%e,%e", p_type, p.a, p.b, p.c);
```

Comment:
I'd be conservative here. What's the order of numbers we are dealing with here?

Reply:
%f format can easily end up with 2 significant digits or less. a is usually the coefficient with the smallest value.
As long as the receiving end uses atof() or scanf ("%f) it'll be safe. Is this a safe assumption??

Reply:
Here is an example:
a: -0.000177, b: 0.032300, c: -2.040927

Reply:
hm, ok, I don't have a strong opinion here because I don't know what these coefficients really mean. The only situation where "e" notation would be a PITA is on embedded code, where you don't have libc, but OTOH we probably don't want to lose precision here.